src/kafka/SendKafkaMeldingTilDatavarehus.kt

```diff
@@ -0,0 +1,24 @@
+package no.nav.rekrutteringsbistand.statistikk.kafka
+
+import io.micrometer.core.instrument.Metrics
+import no.nav.rekrutteringsbistand.statistikk.db.Repository
+import no.nav.rekrutteringsbistand.statistikk.log
+
+fun hentUsendteUtfallOgSendPåKafka(repository: Repository, kafkaProducer: DatavarehusKafkaProducer) = Runnable {
+    val skalSendes = repository.hentUsendteUtfall()
+
+    skalSendes.forEach {
+        repository.registrerSendtForsøk(it)
+        try {
+            kafkaProducer.send(it)
+            repository.registrerSomSendt(it)
+        } catch (e: Exception) {
+            log.error("Prøvde å sende melding på Kafka til Datavarehus om et kandidatutfall", e)
+            Metrics.counter(
+                "rekrutteringsbistand.statistikk.kafka.feilet",
+                "antallSendtForsøk", it.antallSendtForsøk.toString()
+            ).increment()
+            return@forEach
```

Comment:
`return@forEach` er unødvendig her. To alternativer:
1. Slette den. Da logger vi error og metrikk for hvert eneste utfall som feiler, istedenfor å avbryte batchen ved første utfall som feiler.
2. Bytte til `return@Runnable`. Da hopper vi ut av forEach-løkka. Tror jeg.

`return@forEach` fungerer ikke slik som vi trodde: Den bare skipper til neste iterasjon av løkka. Se eksempler [her](https://kotlinlang.org/docs/reference/returns.html) og sitatet "Note that the use of local returns in previous three examples is similar to the use of continue in regular loops."

Reply:
Nice catch!
Jeg bytter til `return@Runnable` slik at vi stopper sending på Kafka hvis én melding feiler.
På den måten vil vi alltid sende meldinger i samme rekkefølge som vi mottok dem.